connd/qconnectionagent.cpp

```diff
@@ -603,90 +624,119 @@ void QConnectionAgent::openConnectionDialog(const QString &type)
 
 void QConnectionAgent::startTethering(const QString &type)
 {
-    if (!netman | (type != "wifi")) { //we only support wifi for now
-        Q_EMIT tetheringFinished(false);
+    if (!netman | (type != "wifi" && type !="bluetooth")) { // support wifi and bt
         return;
     }
-
+    qCDebug(connAgent) << "startTethering" << type;
     NetworkTechnology *tetherTech = netman->getTechnology(type);
     if (!tetherTech) {
-        Q_EMIT tetheringFinished(false);
-        return;
-    }
-
-    QVector <NetworkService *> services = netman->getServices("cellular");
-    if (services.isEmpty()) {
-        Q_EMIT tetheringFinished(false);
-        return;
-    }
-
-    NetworkService *cellService = services.at(0);
-    if (!cellService || netman->offlineMode()) {
-        Q_EMIT tetheringFinished(false);
+        if (type == "wifi") {
+            Q_EMIT wifiTetheringFinished(false);
+        } else {
+            Q_EMIT bluetoothTetheringFinished(false);
+        }
         return;
     }
 
+    bool techPowered = tetherTech->powered();
     QSettings confFile;
     confFile.beginGroup("Connectionagent");
-    bool cellConnected = cellService->connected();
-    bool cellAutoconnect = cellService->autoConnect();
-
-    // save cellular connection state
-    confFile.setValue("tetheringCellularConnected",cellConnected);
-    confFile.setValue("tetheringCellularAutoconnect",cellAutoconnect);
 
-    bool techPowered = tetherTech->powered();
-
-    // save wifi powered state
-    confFile.setValue("tetheringTechPowered",techPowered);
-    confFile.setValue("tetheringType",type);
+    if (type == "wifi") { // Only force cellular on for wifi. Bt can use either when available.
+        QVector <NetworkService *> services = netman->getServices("cellular");
+        if (services.isEmpty()) {
+            Q_EMIT wifiTetheringFinished(false);
+            return;
+        }
+        NetworkService *cellService = services.at(0);
+        if (!cellService || netman->offlineMode()) {
+            Q_EMIT wifiTetheringFinished(false);
+            return;
+        }
+        bool cellConnected = cellService->connected();
+        bool cellAutoconnect = cellService->autoConnect();
+
+        // save cellular connection state
+        confFile.setValue("tetheringCellularConnected",cellConnected);
+        confFile.setValue("tetheringCellularAutoconnect",cellAutoconnect);
```

Comment:
Extra points for adding spaces after commas on these now that you touched them. And the powered below.